driver-core/src/main/java/com/datastax/driver/core/SimpleStatement.java

```diff
@@ -117,7 +114,24 @@ public String getQueryString() {
 
     @Override
     public ByteBuffer[] getValues() {
```

Comment:
The only possible downside I see here is that convert is now computed on each call to getValues().  From what I can tell this is not really that hot of a method (although it is public).  From what I can tell this is only called whenever computing hash for paging state and making the request message, so it does seem like a good idea to compute the values every time it's called as it currently is instead of storing the bytebuffers along with the actual values.